File: Puntocentro.java
package puntocentrolinea;
import javax.swing.*;
import java.awt.*;

class Punto {
    double x;
    double y;

    public Punto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public void coordenadasCartesianas() {
        System.out.println("Los puntos del plano cartesiano son: (" + x + ", " + y + ")");
    }

    public void coordenadasPolares() {
        double c = Math.sqrt(x * x + y * y);
        double a = Math.atan2(y, x);
        System.out.println("Las coordenadas polares son: (r = " + c + ", θ = " + a + ")");
    }

    public String toString() {
        return "(" + x + ", " + y + ")";
    }
}

class Linea extends JPanel {
    Punto p1;
    Punto p2;

    public Linea(Punto p1, Punto p2) {
        this.p1 = p1;
        this.p2 = p2;
        setPreferredSize(new Dimension(5000, 5000));
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        g.setColor(Color.BLUE);
        g.drawLine((int) p1.x, (int) p1.y, (int) p2.x, (int) p2.y);
    }

    public String toString() {
        return "Línea desde el punto " + p1.toString() + " hasta el punto " + p2.toString();
    }

    public void dibujarLinea() {
        System.out.println("Dibujando una línea desde el punto " + p1.toString() + " hasta el punto " + p2.toString());
        JFrame frame = new JFrame("Dibujar Línea");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.add(this);
        frame.pack();
        frame.setVisible(true);
    }
}

class Circulo extends JPanel {
    Punto centro;
    float radio;

    public Circulo(Punto centro, float radio) {
        this.centro = centro;
        this.radio = radio;
        setPreferredSize(new Dimension(5000, 5000));
    }

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        g.setColor(Color.RED);
        int diameter = (int) (radio * 2);
        g.drawOval((int) (centro.x - radio), (int) (centro.y - radio), diameter, diameter);
    }

    public String toString() {
        return "Círculo con centro en los puntos: " + centro.toString() + " y con radio " + radio;
    }

    public void dibujaCirculo() {
        System.out.println("Dibujando un círculo con centro en los puntos " + centro.toString() + " y con radio " + radio);
        JFrame frame = new JFrame("Dibujar Círculo");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.add(this);
        frame.pack();
        frame.setVisible(true);
    }
}

public class PuntoCentroLinea {
    public static void main(String[] args) {
        Punto a = new Punto(11, 3);
        Punto b = new Punto(23, 22);

        System.out.println("Primer punto");
        a.coordenadasCartesianas();
        a.coordenadasPolares();

        System.out.println("Segundo punto");
        b.coordenadasCartesianas();
        b.coordenadasPolares();

        System.out.println("_Línea__");
        Linea l = new Linea(a, b);
        l.toString();
        l.dibujarLinea();

        System.out.println("_Círculo_");
        Circulo c = new Circulo(a, 5.1f);
        c.toString();
        c.dibujaCirculo();

        Circulo d = new Circulo(b, 5.1f);
        d.toString();
        d.dibujaCirculo();
    }
}

File: Puntocentro.py
import math
import tkinter as tk


class Punto:
    def _init_(self, x, y):
        self.x = x
        self.y = y

    def coordenadas_cartesianas(self):
        print(f"Los puntos del plano cartesiano son: ({self.x}, {self.y})")

    def coordenadas_polares(self):
        r = math.sqrt(self.x*2 + self.y*2)
        theta = math.atan2(self.y, self.x)
        print(f"Las coordenadas polares son: (r = {r}, θ = {theta})")

    def _str_(self):
        return f"({self.x}, {self.y})"


class Linea:
    def _init_(self, p1, p2):
        self.p1 = p1
        self.p2 = p2

    def _str_(self):
        return f"Línea desde el punto {self.p1} hasta el punto {self.p2}"

    def dibujar_linea(self, canvas):
        print(f"Dibujando una línea desde el punto {self.p1} hasta el punto {self.p2}")
        canvas.create_line(self.p1.x, self.p1.y, self.p2.x, self.p2.y, fill="blue")


class Circulo:
    def _init_(self, centro, radio):
        self.centro = centro
        self.radio = radio

    def _str_(self):
        return f"Círculo con centro en los puntos: {self.centro} y con radio {self.radio}"

    def dibujar_circulo(self, canvas):
        print(f"Dibujando un círculo con centro en los puntos {self.centro} y con radio {self.radio}")
        canvas.create_oval(
            self.centro.x - self.radio,
            self.centro.y - self.radio,
            self.centro.x + self.radio,
            self.centro.y + self.radio,
            outline="red"
        )


a = Punto(100, 100)
b = Punto(200, 200)


print("Primer punto")
a.coordenadas_cartesianas()
a.coordenadas_polares()

print("Segundo punto")
b.coordenadas_cartesianas()
b.coordenadas_polares()


root = tk.Tk()
root.title("Dibujar Línea y Círculo")
canvas = tk.Canvas(root, width=400, height=400)
canvas.pack()


print("Línea_")
l = Linea(a, b)
print(l)
l.dibujar_linea(canvas)


print("Círculo")
c = Circulo(a, 50)
print(c)
c.dibujar_circulo(canvas)

d = Circulo(b, 50)
print(d)
d.dibujar_circulo(canvas)


root.mainloop()
